### DISCUSSION TITLE
How can we create the LUA terrain part ?

### DISCUSSION THREAD

**Discussion by BigbytheWolfy** [meetric1/gmod-infinite-map#105] [Q&A]
Are there func in a file / documentation that we can use as lua dev to create our own LUA terrain ?
If it's already the case, I apologize, but I didn't saw it.

Great Job by the way, long time I didn't see a genius addon like that, thanks !

**Comment by meetric1** [comment]
Not really, all of the terrain and logic pretty much has to come from you

If you need a (kinda bad) reference, you can check out (or completely copy) files cl_terrain_visual, infmap_collider, infmap_terrain_render, sh_collider_functions, and sv_terrain_collision for my take on some fake lua brush terrain

but yeah, unfortunately infinitely procedurally generated terrain isnt easy

the main problem that I ran into was just how fast the terrain generated. The other logic is mostly not that bad, but it takes for EVER to generate imeshes and physmeshes, especially map wide ones, and of course if you make one too big the game crashes. this is why the default terrain is so low poly
